domain/repositories/champion.go

```diff
@@ -0,0 +1,121 @@
+package repositories
+
+import (
+	"database/sql"
+	"errors"
+
+	"github.com/wizelineacademy/golang-bootcamp-2020/domain/models"
+)
+
+// Errors
+
+// ErrNoRecord is a custom error used in case no Champion is found in the database
+var ErrNoRecord = errors.New("repositories: no matching record found")
+
+// ChampionRepository defines the interface used by a Champion struct to access its repositories methods
+type ChampionRepository interface {
+	GetSingle(id int) (*models.Champion, error)
+	GetMultiple(limit int) ([]*models.Champion, error)
+	Insert(champion *models.Champion) (int, error)
+}
+
+// ChampRepo defines the link between the Champion and the Database
+type ChampRepo struct {
+	DB *sql.DB
+}
+
+// NewChampRepo returns an inituialized ChampRepo struct
+func NewChampRepo(DB *sql.DB) *ChampRepo {
+	return &ChampRepo{DB}
+}
+
+// GetSingle gets a single database row and returns it as a Champion
+func (cr *ChampRepo) GetSingle(id int) (*models.Champion, error) {
+	// SQL statement
+	stmt := `SELECT  name, lore, created FROM champions WHERE id = ?`
+
+	// This returns a pointer to a sql.Row object which holds the result from the database.
+	row := cr.DB.QueryRow(stmt, id)
+
+	// Initialize a pointer to a new zeroed Champion struct.
+	c := &models.Champion{}
+
+	// Copy the values from each field in sql.Row to the corresponding field in the Champion struct.
+	err := row.Scan(&c.Name, &c.Lore, &c.DateCreated)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrNoRecord
+		} else {
+			return nil, err
+		}
+	}
```

Comment:
You could get rid of the whole else block and only leave the return, as your code (at this moment) doesn't have any other possible output.

```suggestion
	if err != nil {
		if errors.Is(err, sql.ErrNoRows) {
			return nil, ErrNoRecord
		} 
		return nil, err
	}
```